src/sagemaker/image_uris.py

```diff
@@ -276,9 +280,9 @@ def _get_image_tag(
 ):
     """Return image tag based on framework, container, and compute configuration(s)."""
     instance_type_family = _get_instance_type_family(instance_type)
-    if (
-        framework in (XGBOOST_FRAMEWORK, SKLEARN_FRAMEWORK)
-        and instance_type_family in GRAVITON_ALLOWED_TARGET_INSTANCE_FAMILY
+    if framework in (XGBOOST_FRAMEWORK, SKLEARN_FRAMEWORK) and (
+        instance_type_family in GRAVITON_ALLOWED_TARGET_INSTANCE_FAMILY
+        or final_image_scope == INFERENCE_GRAVITON
```

Comment:
It might be better to leave the "and" statement to make sure that the customer don't accidentally specify a non-Graviton instance(with the scope flag) and get a cryptic error message? We can instead an added condition to check if the scope param was passed and if it is set to INFERENCE_GRAVITON.

Reply:
Oh good point. I think what we should do then is:

1. If `image_scope == "inference_graviton"` and `instance_type` is `None` or Graviton instance -> return ARM image
2. If `image_scope == "inference_graviton"` and `instance_type` is x86 instance -> raise ValueError

Let me know if this is what you had in mind and if there are any other conditions to catch.

Reply:
I think `image_scope` should not be the unique identifier for the Customers intentions to invoke Graviton instances. Instead can we have a check for `instance_type` when belonging to `GRAVITON_ALLOWED_TARGET_INSTANCE_FAMILY` then invoke the graviton instance and set the scope to `INFERENCE_GRAVITON`? 
This is how it was implemented for PyTorch and Tensorflow frameworks.

Reply:
We can circle back and discuss the implementation. With the existing implementation it is confusing for Customer to use these instances differently for each framework type.